src/graph/visitor/EvaluableExprVisitor.cpp

```diff
@@ -0,0 +1,67 @@
+/* Copyright (c) 2021 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License.
+ */
+
+#include "graph/visitor/EvaluableExprVisitor.h"
+
+#include "graph/context/QueryContext.h"
+
+namespace nebula {
+namespace graph {
+
+EvaluableExprVisitor::EvaluableExprVisitor(const QueryContext *qctx) : qctx_(qctx) {}
+
+void EvaluableExprVisitor::visit(ConstantExpression *) { isEvaluable_ = true; }
+
+void EvaluableExprVisitor::visit(LabelExpression *) { isEvaluable_ = false; }
+
+void EvaluableExprVisitor::visit(UUIDExpression *) { isEvaluable_ = false; }
+
+void EvaluableExprVisitor::visit(VariableExpression *expr) {
+  isEvaluable_ = (qctx_ && qctx_->existParameter(expr->var())) ? true : false;
```

Comment:
```
DCHECK(qctx_)
isEvaluable_ = (qctx_->existParameter(expr->var())) ? true : false;
```

Reply:
`qctx_` is not necessary in the implementation of ngql. Just to be compatible with the previous implementation, the developer can control the intent of the EvaluableExprVisitor.